app/Http/Resources/DownloadResource.php

```diff
@@ -21,7 +21,11 @@ public function toArray($request)
             'icon' => route('applications.icon', $this->application),
             'version' => $this->version,
             'date' => $this->available_from->toDateString(),
-            'download_url' => route('applications.install', [$this->application->slug, $this->platform]),
+            'download_url' => route('applications.install', [
+                $this->application->slug,
+                $this->platform,
+                $this,
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Mmh, i'm a bit for keeping it as it was before, so it will be the job of the install controller figure out the right buid, its true that $this is basically a proxy for the model, but the resource itself doesnt extends eloquent base classes so looks a bit odd (maybe set explicitly the `->id`), what do you think?

Reply:
Yep, much better 👉 https://github.com/everli/uppy/pull/43/commits/149294c200031b578f5a37cfa974064daa9e40a8